armi/reactor/reactors.py

```diff
@@ -531,6 +566,49 @@ def removeAllAssemblies(self, discharge=True):
             self.parent.sfp.removeAll()
         self.blocksByName = {}
         self.assembliesByName = {}
+        self.parent.p.maxAssemNum = 0
+
+    def normalizeNames(self, startIndex=0):
+        """
+        Renumber and rename all the Assemblies and Blocks.
+
+        Parameters
+        ----------
+        startIndex : int, optional
+            The default is to start counting at zero. But if you are renumbering assemblies
+            across the entire Reactor, you may want to start at a different number.
+
+        Returns
+        -------
+        int
+            The new max Assembly number.
+        """
+        ind = startIndex
+        for a in self:
+            oldName = a.getName()
+            newName = a.makeNameFromAssemNum(ind)
+            if oldName == newName:
+                ind += 1
+                continue
+
+            a.p.assemNum = ind
+            a.setName(newName)
+
+            for b in a:
+                axialIndex = int(b.name.split("-")[-1])
+                b.name = b.makeName(ind, axialIndex)
+
+            ind += 1
+
+        # Update some bookkeeping dictionaries of assembly and block names in this Core.
+        self.assembliesByName = {}
+        self.blocksByName = {}
+        for assem in self:
+            self.assembliesByName[assem.getName()] = assem
+            for b in assem:
+                self.blocksByName[b.getName()] = b
```

Comment:
Can we move these operations into the loop above? That would help us to avoid calling `*.getName()` many extra times.

Reply:
I thought this was easier to read.

In fact, I was thinking of moving this "re-grab the names" block of code and moving it to a helper method.

You don't think it would  be an uglier loop the other way?  I guess I could do that, yeah.

Reply:
I personally think the other way would be nicer to read, as it puts related operations closer to each other.